plugins/kotlin/src/main/java/org/quiltmc/quiltflower/kotlin/struct/KProperty.java

```diff
@@ -0,0 +1,359 @@
+package org.quiltmc.quiltflower.kotlin.struct;
+
+import kotlin.reflect.jvm.internal.impl.metadata.ProtoBuf;
+import kotlin.reflect.jvm.internal.impl.metadata.jvm.JvmProtoBuf;
+import org.jetbrains.annotations.Nullable;
+import org.jetbrains.java.decompiler.code.CodeConstants;
+import org.jetbrains.java.decompiler.main.ClassesProcessor;
+import org.jetbrains.java.decompiler.main.rels.ClassWrapper;
+import org.jetbrains.java.decompiler.main.rels.MethodWrapper;
+import org.jetbrains.java.decompiler.modules.decompiler.exps.ConstExprent;
+import org.jetbrains.java.decompiler.modules.decompiler.exps.Exprent;
+import org.jetbrains.java.decompiler.struct.StructClass;
+import org.jetbrains.java.decompiler.struct.StructField;
+import org.jetbrains.java.decompiler.struct.StructMethod;
+import org.jetbrains.java.decompiler.struct.attr.StructConstantValueAttribute;
+import org.jetbrains.java.decompiler.struct.attr.StructGeneralAttribute;
+import org.jetbrains.java.decompiler.struct.consts.PrimitiveConstant;
+import org.jetbrains.java.decompiler.struct.gen.VarType;
+import org.jetbrains.java.decompiler.util.InterpreterUtil;
+import org.jetbrains.java.decompiler.util.TextBuffer;
+import org.jetbrains.java.decompiler.util.collections.VBStyleCollection;
+import org.quiltmc.quiltflower.kotlin.KotlinDecompilationContext;
+import org.quiltmc.quiltflower.kotlin.KotlinPreferences;
+import org.quiltmc.quiltflower.kotlin.KotlinWriter;
+import org.quiltmc.quiltflower.kotlin.metadata.MetadataNameResolver;
+import org.quiltmc.quiltflower.kotlin.util.KTypes;
+import org.quiltmc.quiltflower.kotlin.util.ProtobufFlags;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+
+public class KProperty {
```

Comment:
Would be better if it were final. Also, this class is rather complicated and so justifies the need for more code comments throughout.